precompiles/src/solidity/codec/bytes.rs

```diff
@@ -68,6 +68,20 @@ impl<K: Kind, S: Get<u32>> Clone for BoundedBytesString<K, S> {
 	}
 }
 
+impl<K1, S1, K2, S2> PartialEq<BoundedBytesString<K2, S2>> for BoundedBytesString<K1, S1> {
```

Comment:
Any differences between this and the derived implementation?

Reply:
The Derive macro tries to implement them by comparing all fields but PhantomData currently does not work with the derive macro due to incorrect bounds. The issue is tracked here https://github.com/rust-lang/rust/issues/26925